lib/plugins/aws/tests/naming.js

```diff
@@ -0,0 +1,435 @@
+    options = {
+      stage: 'dev',
+      region: 'us-east-1',
+    };
+    serverless = new Serverless(options);
+    sdk = new SDK(serverless, options);
+  });
+
+  it('#normalizeName()', () => {
+    it('should capitalize the first letter', () => {
+      expect(sdk.naming.normalizeName('name')).to.equal('Name');
+    });
+    it('should have no effect on caps', () => {
+      expect(sdk.naming.normalizeName('Name')).to.equal('Name');
+    });
+    it('should have no effect on the rest of the name', () => {
+      expect(sdk.naming.normalizeName('nAME')).to.equal('NAME');
+    });
+  });
+  it('#normalizeNameToAlphaNumericOnly()', () => {
+    it('should strip non-alpha-numeric characters', () => {
+      expect(sdk.naming
+        .normalizeNameToAlphaNumericOnly('`!@#$%^&*()-={}|[]\\:";\'<>?,./')).to.be('');
+    });
+    it('should apply normalizeName to the remaining characters', () => {
+      expect(sdk.naming.normalizeNameToAlphaNumericOnly('a-b-c')).to.be('Abc');
+    });
+  });
+  it('#normalizeNameToCapitalAlphaNumbericOnly()', () => {
+    it('converts `-` to `Dash`', () => {
+      expect(sdk.naming.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(
+        'a-path'
+      )).to.equal('ADashpath');
+    });
+    it('converts variable declarations (`${var}`) to `1Var`', () => {
+      expect(sdk.naming.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(
+        '${variable}'
+      )).to.equal('1Var');
+    });
+    it('converts variable declarations prefixes to `1Var`', () => {
+      expect(sdk.naming.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(
+        '${variable}Path'
+      )).to.equal('1VarPath');
+    });
+    it('converts variable declarations suffixes to `1Var`', () => {
+      expect(sdk.naming.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(
+        'path${variable}'
+      )).to.equal('Path1Var');
+    });
+    it('converts variable declarations in center to `1Var`', () => {
+      expect(sdk.naming.normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(
+        'path${variable}Dir'
+      )).to.equal('Path1VarDir');
+    });
+  });
+  it('#getServiceEndpointRegex()', () => {
+    it('should match the prefix', () => {
+      expect(sdk.naming.getServiceEndpointRegex().match('ServiceEndpoint')).to.be(true);
+    });
+    it('should not match a name without the prefix', () => {
+      expect(sdk.naming.getServiceEndpointRegex().match('NotThePrefixServiceEndpoint')).to.be(true);
+    });
+    it('should match a name with the prefix', () => {
+      expect(sdk.naming.getServiceEndpointRegex().match('ServiceEndpointForAService')).to.be(true);
+    });
+  });
+  it('#getStackName()', () => {
+    it('should use the service name and stage from the service and config', () => {
+      serverless.service.service = 'myService';
+      expect(sdk.naming.getStackName()).to.equal(`${serverless.service.service}-${options.stage}`);
+    });
+    it('should handle undefined service name', () => {
```

Comment:
Should it really handle `undefined` or throw an error instead? (same for the `stage` etc.)